spring-ai-client-chat/src/main/java/org/springframework/ai/chat/client/advisor/MessageChatMemoryAdvisor.java

```diff
@@ -43,64 +40,49 @@ public MessageChatMemoryAdvisor(ChatMemory chatMemory) {
 		super(chatMemory);
 	}
 
-	public MessageChatMemoryAdvisor(ChatMemory chatMemory, String defaultConversationId, int chatHistoryWindowSize) {
-		this(chatMemory, defaultConversationId, chatHistoryWindowSize, Advisor.DEFAULT_CHAT_MEMORY_PRECEDENCE_ORDER);
+	public MessageChatMemoryAdvisor(ChatMemory chatMemory, String defaultConversationId) {
+		this(chatMemory, defaultConversationId, Advisor.DEFAULT_CHAT_MEMORY_PRECEDENCE_ORDER);
 	}
 
-	public MessageChatMemoryAdvisor(ChatMemory chatMemory, String defaultConversationId, int chatHistoryWindowSize,
-			int order) {
-		super(chatMemory, defaultConversationId, chatHistoryWindowSize, true, order);
+	public MessageChatMemoryAdvisor(ChatMemory chatMemory, String defaultConversationId, int order) {
+		super(chatMemory, defaultConversationId, true, order);
 	}
 
 	public static Builder builder(ChatMemory chatMemory) {
 		return new Builder(chatMemory);
 	}
 
 	@Override
-	public ChatClientResponse adviseCall(ChatClientRequest chatClientRequest, CallAdvisorChain callAdvisorChain) {
-		chatClientRequest = this.before(chatClientRequest);
-
-		ChatClientResponse chatClientResponse = callAdvisorChain.nextCall(chatClientRequest);
-
-		this.after(chatClientResponse);
-
-		return chatClientResponse;
-	}
-
-	@Override
-	public Flux<ChatClientResponse> adviseStream(ChatClientRequest chatClientRequest,
-			StreamAdvisorChain streamAdvisorChain) {
-		Flux<ChatClientResponse> chatClientResponses = this.doNextWithProtectFromBlockingBefore(chatClientRequest,
-				streamAdvisorChain, this::before);
-
-		return new MessageAggregator().aggregateChatClientResponse(chatClientResponses, this::after);
+	public ChatClientRequest before(ChatClientRequest request, AdvisorChain advisorChain) {
+		String conversationId = doGetConversationId(request.context());
+		// Add the new user messages from the current prompt to memory
+		List<UserMessage> newUserMessages = request.prompt().getUserMessages();
```

Comment:
As we discussed, I'm not convinced about this logic because there can be more than user messages in the input list. 

If we want to handle it after RC1 and before the GA version, I would consider keeping the logic we had before here, with just the last `UserMessage` being included in the memory. 

What do you think?

Reply:
For example, something I do a lot, is passing as input messages a list of `UserMessage`/`AssistantMessage` pairs as few shots. With the current logic, I would put all the `UserMessage`s in memory, but not the `AssistantMessage`s.